qemu/qmp.go

```diff
@@ -1255,6 +1255,44 @@ func (q *QMP) ExecHotplugMemory(ctx context.Context, qomtype, id, mempath string
 	return err
 }
 
+// ExecuteNVDIMMDeviceAdd adds the guest portion of a block device to
+// a QEMU instance using a NVDIMM driver with the device_add command.
+// id is the id of the device to add.  It must be valid a QMP identifier.
+// mempath is the path of the device to add, e.g., /dev/rdb0.  size is
+// the data size of the device.
+func (q *QMP) ExecuteNVDIMMDeviceAdd(ctx context.Context, id, mempath string, size int64) error {
+	args := map[string]interface{}{
+		"qom-type": "memory-backend-file",
+		"id":       "nvdimmbackmem" + id,
+		"props": map[string]interface{}{
+			"mem-path": mempath,
+			"size":     size,
+			"share":    true,
+		},
+	}
+	err := q.executeCommand(ctx, "object-add", args, nil)
+	if err != nil {
+		return err
+	}
+
+	args = map[string]interface{}{
+		"driver": "nvdimm",
+		"id":     "nvdimm" + id,
+		"memdev": "nvdimmbackmem" + id,
+	}
+	if err = q.executeCommand(ctx, "device_add", args, nil); err != nil {
+		if err != nil {
```

Comment:
This if statement is not needed.  If we get here we err !=nil as this condition is tested on the previous line.